### ISSUE TITLE
[BUG] A modified note will appear in "LIST CREATED" log, with a wrong timecode.

### ISSUE THREAD

**Issue by Elaws** [franciskafieh/obsidian-list-modified#120]
## Describe the bug

Desktop (Windows 11) and mobile (Android) are synced through SyncTrayzor/Syncthing. I modified a note for the first time of the day on Obsidian desktop (Obsidian mobile is closed and was not opened in the last 2 hours). 

## Expected behaviour

The modification should be logged in "LIST MODIFIED".

## Actual behavior

For some reason, that modification created a log in "LIST CREATED", with a wrong timecode (note was modified around 00:30AM, and timecode is 11:33AM).

Please find below a screenshot of daily notes from 11th of November (yesterday), and from 12th of November (today, when the note was modified).

11th November daily note :

![image](https://github.com/user-attachments/assets/52c27a42-f534-478c-b2da-a3f28ff1744f)

12th November daily note :

![image](https://github.com/user-attachments/assets/23a5d311-56aa-483d-83ff-b2fae1b53c3a)

### My setup

- Windows 11
- Obsidian 1.7.5
- List Modified v.3.2.0
- SyncTrayzor 1.28.0
- Android device with Obsidian and Syncthing app.

### Steps to reproduce bug

To be investigated.

**Comment by franciskafieh** [comment]
The modification time is taken directly from Obsidian's file info `mtime` property, which is known to mess up with sync. A possible alternative would be to use a plugin that adds your current time to a property when you modify the file, and then use custom List Modified placeholders in your output format (e.g. [[f.modifytime]]). I think [this plugin](https://github.com/beaussan/update-time-on-edit-obsidian) may do this, but I am not completely sure after reading the readme. There is a plugin out there that does not count on Obsidian's file times, but I can't remember the name atm.

> The modification should be logged in "LIST MODIFIED".

I will look into this.

**Comment by Elaws** [comment]
If it may help, I noticed that the first note I modify for the first time of the day and which creates a log in "LIST CREATED", comes with the following warning :

![image](https://github.com/user-attachments/assets/e611e7e8-78f1-46fb-b8ae-e7b8af141b0f)

Whatever note I modify for the rest of the day won't create an entry in "LIST CREATED". It seems to be only the first modification of the first note of the day.

**Comment by franciskafieh** [comment]
This is the same issue as #114 . Have you updated the plugin yet? The release from 4 days ago fixes that issue. Let me know how that goes

**Comment by technotherion** [comment]
> The modification time is taken directly from Obsidian's file info `mtime` property, which is known to mess up with sync. A possible alternative would be to use a plugin that adds your current time to a property when you modify the file, and then use custom List Modified placeholders in your output format (e.g. [[f.modifytime]]). I think [this plugin](https://github.com/beaussan/update-time-on-edit-obsidian) may do this, but I am not completely sure after reading the readme. There is a plugin out there that does not count on Obsidian's file times, but I can't remember the name atm.
> 
> > The modification should be logged in "LIST MODIFIED".
> 
> I will look into this.

In case anyone else is looking for a plugin that has this functionality, I use Linter which has auto-updating YAML tag functions to set my created_date and modified_date properties. Linter is one of the plugins that I recommend most strongly of any plugin I've used.

Speaking of which, I am using these frontmatter properties ([[f.created_date]] and [[f.modified_date]]) in order to show the correct date and time since I am using a sync solution. However, this does not exclude notes that have an incorrect mtime, so I get weekly note entries that look like the following: (this is a small sample) 

```
- [[2024-11-15]] - 2024-11-15 01:39:38
- [[Ghost Ref]] - 2024-11-15 01:59:33
- [[4. Archives/2. Areas/Career/Portfolio]] - 2023-08-23 10:00:00
- [[Imp Ori]] - 2024-09-05 01:42:47
- [[Representing times of day]] - 2024-02-29 12:00:00
- [[Car Tool Box]] - 2024-09-01 02:29:35
- [[Car Cleaning Kit]] - 2024-07-06 12:21:49
- [[Battery Replacement]] - 2024-09-12 01:34:36
- [[Jeep Not starting - Battery]] - 2024-07-24 09:14:26
- [[Toys]] - 2024-09-07 12:50:00
- [[Clawssss]] - 2024-09-29 12:37:12
- [[Booster Seat idea]] - 2024-08-22 01:05:15
- [[wish list]] - 2024-07-21 12:09:25
- [[Why You Should Flatten Your Document Libraries]] - 2024-11-16 06:04:10
- [[Linking to Headings and Blockquotes]] - 2024-11-16 07:10:45
- [[! Capture/2024-11-16]] - 2024-11-16 10:27:55
```

The common link between the files from other weeks/months that show on my Created list is that they are files that I've moved to my "4. Archive" directory. I have a plugin that automatically changes the path of the files when I archive them. Maybe this is tricking the plugin into thinking I've created new files instead of moving them?

**Comment by franciskafieh** [comment]
> Thank you ! Will update / close if needed.

Any updates?


> I have a plugin that automatically changes the path of the files when I archive them. Maybe this is tricking the plugin into thinking I've created new files instead of moving them?

If a file is renamed (incl. changed path), List Modified will add it to the created list if it is not already tracked. The only workaround for this would be to add your `4. Archive` directory to the excluded directions in List Modified settings.

**Comment by Elaws** [comment]
I'm afraid I was able to reproduce the problem right now. I modified a note (that was created last sunday) past midnight for the first time of the day (20th of November), and that note has been added to the "LIST CREATED" log of today.

So that note appears in 2 daily notes : the daily note of 17th of November, where the note was created, and the daily note of 20th of November, with a wrong timecode (it says 5:32PM instead of shortly after midnight).

**Comment by franciskafieh** [comment]
This is also intentional. If you want to disable this, please enable the
"combine created and modified" setting.

**Comment by Elaws** [comment]
@franciskafieh : I added something in my last message, after `EDIT :`, that will help debug this issue.

> This is also intentional. If you want to disable this, please enable the
"combine created and modified" setting.
[…](https://github.com/franciskafieh/obsidian-list-modified/issues/120#)

Mmm... thank you, however I'd like to keep track of both when I create a note **and** when I modify the note. An option to still track modified note within the same day would be very useful (so note would appear both in `LIST CREATED` and `LIST MODIFIED` only if it is modified after in the same day).

**Comment by franciskafieh** [comment]
with the way the plugin currently works having a note show in 2 lists would
be very difficult to implement. As a workaround, maybe turn on combine
created and modified and use ctime template? / i could make a template
[[createdToday]] that is replaced with “created today” or something if the
ctime is today. Let me know what you think

Otherwise, thanks for making me aware of your edit as I did not see it
through email. Again, will look into it ASAP but that is very helpful

**Comment by Elaws** [comment]
> with the way the plugin currently works having a note show in 2 lists would
be very difficult to implement. As a workaround, maybe turn on combine
created and modified and use ctime template?

Do you mean I use the `Combine created and modified` option and use e.g. `[[link]] last modified [[mtime]] and created [[ctime]]` in `Modified output format` option ? If so, I think I prefer having `LIST CREATED` and `LIST MODIFIED` as separated lists, for clarity purpose. 

On a second thought, having the note only appear in the `LIST CREATED` log and not in the `LIST MODIFIED` log if it is modified later the same day maybe isn't this big of an issue, as long as it does correctly appear in the `LIST MODIFIED` log the next days (which is the issue discussed here ^^).

Anyway, I'll think about it and experiment with your suggestion, thank you !

> / i could make a template
[[createdToday]] that is replaced with “created today” or something if the
ctime is today. Let me know what you think

I'm sorry I don't understand how it differs from your previous suggestion.

> Otherwise, thanks for making me aware of your edit as I did not see it
through email. Again, will look into it ASAP but that is very helpful

Thank you, looking forward to hearing news about it !

**Comment by franciskafieh** [comment]
> I'm sorry I don't understand how it differs from your previous suggestion.

Instead of a timestamp, it could be a checkmark emoji or say "created today" or something

**Comment by Elaws** [comment]
> > Today, I modified the note again, and it appears in LIST CREATED of today. Here is the daily note from today :
> 
> Okay, so from a quick re-read, this seems to be the main issue here. This is unexpected. If the note was not created by a user, it should show as modified instead. I am guessing this is a sync issue. Have you tried disabling the plugin one all devices but one? Does this fix the issue?

Hi, the plugin parameter `isDesktopOnly` is set to `true`, so that it is only enabled on my computer (Windows 11), and not my Android phone neither iPad.

**Comment by franciskafieh** [comment]
Got it. Does this issue only occur when modifying from a device other than your computer? This would make sense:

Computer is the only device with Obsidian List Modified. If you create a file on your phone, for example, and it syncs to your computer, in your computer's eyes, this file was newly created. Does this make sense?

Please let me know if this is the issue. Sync is a very difficult topic to test, and I am not 100% sure how I could implement a solution for this.

**Comment by Elaws** [comment]
> Got it. Does this issue only occur when modifying from a device other than your computer? This would make sense:
> 
> Computer is the only device with Obsidian List Modified. If you create a file on your phone, for example, and it syncs to your computer, in your computer's eyes, this file was newly created. Does this make sense?
> 
> Please let me know if this is the issue. Sync is a very difficult topic to test, and I am not 100% sure how I could implement a solution for this.

No, the problem described here happens when modifying an already existing file on my desktop (Windows 11). I've not yet tried to modify an already existing note on Android or iPad.

**Comment by Elaws** [comment]
Another argument for this bug not being caused by a sync issue, is that only an already existing not that is modified **for the first time of the day** is concerned by this bug.

All already existing notes that are modified after that one behave properly.

**Comment by Elaws** [comment]
Bug can be reproduced with 3.2.2.

I created a note yesterday (9th of December), here is the daily note of  December, 9th : 

![image](https://github.com/user-attachments/assets/f3a0a7a3-3c85-44ab-9b2c-339723a9a6de)

Made a modification today (10th of December), here is the daily note of December, 10th : 

![image](https://github.com/user-attachments/assets/7978476d-b453-47be-9d0f-c8b5a2d3f470)

Any other modifications made to already existing notes appear properly in "LIST MODIFIED". Only the first modification of the day, of an already existing note, makes the note appear in "LIST CREATED" instead of "LIST MODIFIED" log.

**Comment by Elaws** [comment]
I forgot to add that as always, the bug is accompanied by this message in the console :

![image](https://github.com/user-attachments/assets/8f04bf3b-f54b-4b0e-9e53-53c80a7951f6)

**Comment by franciskafieh** [comment]
Thank you. Give me three more days and I promise I will have time to look into it :joy: school break starts soon

**Comment by franciskafieh** [comment]
Hi there. I finally have some time to look into this. I still cannot re-produce this on my computer, so sorry for all the questions. 

> the bug is accompanied by this message in the console :
> 
> ![image](https://private-user-images.githubusercontent.com/52013479/395806830-2ca6eb85-3b72-497c-bcba-61c7c6b7f853.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NDA3ODIsIm5iZiI6MTczNDc0MDQ4MiwicGF0aCI6Ii81MjAxMzQ3OS8zOTU4MDY4MzAtMmNhNmViODUtM2I3Mi00OTdjLWJjYmEtNjFjN2M2YjdmODUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDAwMjEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxZjI1MDQ1MjRlNDkzYzVjMDg1YzZiMzcyZTg1NjhkOGY1Yjc2ZjFhNGRmY2NiZjA2NmQ2MTZiNWQxMDg0MTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RUpthduXh6VClg053sYaCHn6j4OP2zCQ6Y1SvWajuoY)

This is normal for new day/week/month logic, which is consistent with

> Only the first modification of the day, of an already existing note, makes the note appear in "LIST CREATED" instead of "LIST MODIFIED" log.



Can you please turn on verbose/debug mode in the Obsidian List Modified settings?

The next time this bug occurs, right before the output you have posted, there will be a message saying ""New period logic is being ran by [something here]"

Supposedly, this would say created but should say modified.

Thank you

**Comment by Elaws** [comment]
Please find below the requested information.

As soon as I turned on verbose mode, I got what follows :

![image](https://github.com/user-attachments/assets/1eacf071-8689-4fee-8eee-7969e8a6e8a0)

After modifying an already existing note, I got what follows, however the note was properly put in "LIST MODIFIED", and not "LIST CREATED". 

![image](https://github.com/user-attachments/assets/514000a2-9c84-44db-8a8e-0dabf84e7d24)

It's 23:16 here, I think this is the first already existing note I modified for the first time today, so the bug should have happened. I will try again past midnight and update.

**Comment by Elaws** [comment]
Okay so I was able to reproduce the bug, by modifying an already existing note (`) Histoire de l'Europe`), for the first time of the day, around 00:05.

When launching Obsidian, I got the following :

![image](https://github.com/user-attachments/assets/19db7f2b-9e89-4129-affb-27015f196389)

When modifying (`) Histoire de l'Europe`), I got :

![image](https://github.com/user-attachments/assets/df9bc63b-88ae-4855-a05a-f01956e227c0)

![image](https://github.com/user-attachments/assets/dc7f69fa-ddb6-40e1-9e7b-ce32a4275f56)

**Comment by franciskafieh** [comment]
![image](https://github.com/user-attachments/assets/51e084e3-ffb4-4378-8d82-dd458d7ce096)
according to logs, this should be marked as modified, as expected. Interesting. let me look into this.

**Comment by franciskafieh** [comment]
should be fixed in 3.3.0. Re-open otherwise. Thanks again for the report!

**Comment by Elaws** [comment]
This is great news, thank you !
